app/server/utils/user.py

```diff
@@ -858,43 +864,43 @@ def attach_transfer_account_to_user(user, organisation=None):
     The method also takes an organization object as an argument to match the user object to a specific organization.
 
     :param user: The initial account the user created when they began the self sign up process.
-    :param organisation: The organization the user belongs to.
     :return: A user with a transfer account,
     """
+
+    if user.primary_blockchain_address:
+        raise Exception('User already has a transfer account attached.')
+
+    organisation = g.active_organisation
     if not organisation:
-        organisation = Organisation.query.get(2)
+        raise Exception("No active organisation")
 
-    if user.primary_blockchain_address is None:
-        blockchain_address = bt.create_blockchain_wallet()
-        user.primary_blockchain_address = blockchain_address
-        user.set_held_role('BENEFICIARY', 'beneficiary')
-        user.add_user_to_organisation(organisation, is_admin=False)
+    blockchain_address = bt.create_blockchain_wallet()
+    user.primary_blockchain_address = blockchain_address
+    user.set_held_role('BENEFICIARY', 'beneficiary')
+
+    db.session.add(user)
 
-        db.session.add(user)
+    transfer_account = TransferAccount(bound_entity=user,
+                                       organisation=organisation,
+                                       blockchain_address=blockchain_address)
+    db.session.add(transfer_account)
+    user.default_transfer_account = transfer_account
 
-        transfer_account = TransferAccount(bound_entity=user,
-                                           organisation=organisation,
-                                           blockchain_address=blockchain_address)
-        db.session.add(transfer_account)
-        user.default_transfer_account = transfer_account
-
-        org_users = organisation.users
-        initial_disbursement = config.SELF_SERVICE_WALLET_INITIAL_DISBURSEMENT
-        disbursing_admin = None
-        for org_user in org_users:
-            if org_user.has_admin_role:
-                disbursing_admin = org_user
-                break
-        if disbursing_admin:
-            initial_disbursement = make_payment_transfer(transfer_amount=initial_disbursement,
-                                                         send_user=disbursing_admin,
-                                                         receive_user=user,
-                                                         transfer_subtype=TransferSubTypeEnum.DISBURSEMENT)
-            db.session.add(initial_disbursement)
-        else:
-            raise Exception('No admin user was found.')
-    else:
-        raise Exception('User already has a transfer account attached.')
+    org_users = organisation.users
+    initial_disbursement = config.SELF_SERVICE_WALLET_INITIAL_DISBURSEMENT
+    disbursing_admin = None
+    for org_user in org_users:
+        if org_user.has_admin_role:
```

Comment:
this feels a bit weird to me - as it means we're selecting an admin user without rhyme or reason and just assigning the transfer to them. It should be relatively straightforward to assign the transfer to the org level transfer account.

Also at worst, this loops through the entire user list for an org, which is now like 20k items!!

Reply:
I agree this is very weird. I commented this on the previous PR too. If "assign the transfer to the org level transfer account" is complicated, then at a very minimum of improvement we should order the `org_users` by `id`? as the lower numbers will have admin status.

Reply:
But it breaks at the point it finds one